### PR TITLE
AccessControl: Grant data source reader to all users when running oss

### PR DESCRIPTION
**Pull request by kalleep**
**What this PR does / why we need it**:
When running oss or enterprise without a license all users should be able to query data soruces.
The previous workaround for this was to grant views with only the action `datasources:query` and all checks where done in enterprise code.

Alerting now also relies on data source permissions and to solve this we can seed the `fixed:datasources:reader` role to all veiwers except in the case where the instance is running enterprise version with a valid or expired license

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
* Also removed the compatibility role we had before

### CODE REVIEW

**Comment by yuri-tceretian** [comment]
I confirm that this fixes the access for Viewer and Editor roles in Grafana OSS with enabled RBAC.

File: pkg/api/accesscontrol.go [hunk 1]
```diff
@@ -419,8 +408,8 @@ func (hs *HTTPServer) declareFixedRoles() error {
 	}
 
 	return hs.AccessControl.DeclareFixedRoles(
-		provisioningWriterRole, datasourcesReaderRole, datasourcesWriterRole, datasourcesIdReaderRole,
-		datasourcesCompatibilityReaderRole, orgReaderRole, orgWriterRole,
+		provisioningWriterRole, datasourcesReaderRole, datasourcesWriterRole,
+		datasourcesIdReaderRole, orgReaderRole, orgWriterRole,
```
Comment (gamab) [inline pkg/api/accesscontrol.go:412]: Do we need a migration to remove the role and its grants from the database? 
Seeding should take care of removing the permissions from the basic roles natively.

Comment (kalleep) [reply pkg/api/accesscontrol.go:412]: Yes i seems like we have to remove that role, We could do it in another pr

File: pkg/api/accesscontrol.go [hunk 2]
```diff
@@ -94,6 +94,10 @@ func (hs *HTTPServer) declareFixedRoles() error {
 		Grants: []string{string(models.ROLE_ADMIN)},
 	}
 
+	if !hs.License.FeatureEnabled("accesscontrol.enforcement") {
```
Comment (gamab) [inline pkg/api/accesscontrol.go:97]: Nit. A little comment might be useful

File: pkg/api/common_test.go [hunk 3]
```diff
@@ -11,6 +11,8 @@ import (
 	"path/filepath"
 	"testing"
 
+	"github.com/grafana/grafana/pkg/services/licensing"
```
Comment (gamab) [inline pkg/api/common_test.go:14]: Nit. :arrow_down:

File: pkg/tests/api/alerting/api_alertmanager_test.go [hunk 4]
```diff
@@ -2454,8 +2454,8 @@ func TestEval(t *testing.T) {
 				}
 			}
 			`,
-			expectedStatusCode: http.StatusUnauthorized,
-			expectedMessage:    "user is not authorized to query one or many data sources used by the rule",
+			expectedStatusCode: http.StatusBadRequest,
+			expectedMessage:    "invalid condition: invalid query A: data source not found: unknown",
```
Comment (IevaVasiljeva) [inline pkg/tests/api/alerting/api_alertmanager_test.go:2458]: question: are these test changes related to the other changes or a separate fix?

**Review by gamab — COMMENTED** [review]
LGTM. 
I'm going to let someone from the alerting squad (@yuri-tceretian?) say whether it's ok to have `authorizeDatasourceAccessForRule` always returning `true`  for open source users.

It's a good thing that we removed the compatibility reader :+1: 
Quick remarks:
- [ ] Can we add scopes to the evaluators now: https://github.com/grafana/grafana/blob/85918202bcb869233b89f70aed61cc4296f40e32/pkg/api/api.go#L340-L352 (other PR?)
- [x] Should we update docs to remove the compatibility reader (other PR?)

**Review by IevaVasiljeva — COMMENTED** [review]
One scenario that worries me is customer moving from OSS to enterprise Grafana. From what I understand, all the viewers and editors would then have access to read and query all data sources? Or would the seeder remove this assignment?

**Comment by yuri-tceretian** [comment]
> I'm going to let someone from the alerting squad (@yuri-tceretian?) say whether it's ok for open source users to have authorizeDatasourceAccessForRule always returning true.

Yes, that is the desired behavior.

**Comment by IevaVasiljeva** [comment]
> > One scenario that worries me is customer moving from OSS to enterprise Grafana. From what I understand, all the viewers and editors would then have access to read and query all data sources? Or would the seeder remove this assignment?
> 
> When running oss fixed roles are not stored in the database so this is not a issue

Ah, true that. In that case this looks good to me!

**Review by IevaVasiljeva — APPROVED** [review]

**Review by gamab — APPROVED** [review]

**Review by gamab — CHANGES_REQUESTED** [review]
Sorry :confused: I forgot to check one thing. Are we ok to enable `viewers` to have full access on the following endpoints which used to be `orgAdmin` only:
https://github.com/grafana/grafana/blob/85918202bcb869233b89f70aed61cc4296f40e32/pkg/api/api.go#L306
https://github.com/grafana/grafana/blob/85918202bcb869233b89f70aed61cc4296f40e32/pkg/api/api.go#L313-L315

I seem to recall something about the list being available in the frontend settings anyway?

**Review by gamab — APPROVED** [review]
Never mind, I went ahead and double checked `filterDsByQueryPermission`. This PR is not introducing any change in behavior in regards to the `GetDatasources` endpoints. All good :+1:

**Comment by yuri-tceretian** [comment]
As long as this fix did not land in beta1 we probably need some migration to happen for those who installed beta1 in order to fix the problem
